jax/_src/numpy/reductions.py

```diff
@@ -90,6 +90,13 @@ def check_where(name: str, where: ArrayLike | None) -> Array | None:
     )
   return where
 
+def check_mean(name: str, mean: ArrayLike | None) -> Array | None:
+  if mean is None:
+    return None
+  mean = ensure_arraylike(name, mean)
+  if not dtypes.issubdtype(mean.dtype, np.inexact):
```

Comment:
I don't think we need to check the dtype of the mean: we should just cast it to the necessary dtype within the computation.

An important aspect that we *do* need to check is the requirement that the mean has the expected shape: i.e. the same shape as `a.mean(axes=axes, keepdims=True)`. If that property does not hold, then we'll either get an error or unexpected results.